Mathlib/Data/Zmod/Defs.lean

```diff
@@ -46,37 +46,64 @@ to register the ring structure on `zmod n` as type class instance.
 open Nat.ModEq Int
 
 /-- Multiplicative commutative semigroup structure on `fin n`. -/
-instance (n : ℕ) : CommSemigroup (Fin n) :=
-  {
-    Fin.hasMul with
-    mul_assoc := fun ⟨a, ha⟩ ⟨b, hb⟩ ⟨c, hc⟩ =>
-      Fin.eq_of_veq
-        (calc
-          a * b % n * c ≡ a * b * c [MOD n] := (Nat.mod_modEq _ _).mul_right _
-          _ ≡ a * (b * c) [MOD n] := by rw [mul_assoc]
-          _ ≡ a * (b * c % n) [MOD n] := (Nat.mod_modEq _ _).symm.mul_left _
-          )
-    mul_comm := Fin.mul_comm }
-
-private theorem left_distrib_aux (n : ℕ) : ∀ a b c : Fin n, a * (b + c) = a * b + a * c :=
-  fun ⟨a, ha⟩ ⟨b, hb⟩ ⟨c, hc⟩ =>
-  Fin.eq_of_veq
-    (calc
-      a * ((b + c) % n) ≡ a * (b + c) [MOD n] := (Nat.mod_modEq _ _).mul_left _
-      _ ≡ a * b + a * c [MOD n] := by rw [mul_add]
-      _ ≡ a * b % n + a * c % n [MOD n] := (Nat.mod_modEq _ _).symm.add (Nat.mod_modEq _ _).symm
-      )
-#align fin.left_distrib_aux fin.left_distrib_aux
```

Comment:
I think you should restore these lines, and the old `CommRing` instance

Reply:
There were some errors in the generated code, so it seemed easier to use the working code. I can take a look at fixing them instead, though probably after `CommRing` works

Reply:
> I can take a look at fixing them instead, though probably after CommRing works

I don't understand this comment; the code that I'm suggesting you throw away is the CommRing code you wrote / took from the ad-hoc port. We should put the mathport CommRing code back and then try and fix the errors it creates.

Reply:
Removing curly braces at line 50 and 59 worked when I was working on 
[commit fd60ee036a8a9b33183779cce999dec6f24f43d5](https://github.com/leanprover-community/mathlib4/pull/1684/commits/fd60ee036a8a9b33183779cce999dec6f24f43d5).

Reply:
Oh, I'm sorry for the confusion. It's `CommRing (Zmod n)` that's the main blocker here, not anything about `Fin`

Reply:
I've pushed a fix to that instance